### ISSUE TITLE
Can't remove an asset on a translated entry

### ISSUE THREAD

**Issue by tao** [statamic/cms#3738]
## Bug Description
When you delete an asset on a translated entry in the control panel, it sets the value to null. This is related to #3406.

However the asset is still returning the original english value. You can see from the screenshots that the bulgarian entry has a null value for the pdf, but in the `{{ dump }}` the english PDF is returning instead... where this should be null instead (or removed from the response?)

![Screenshot 2021-05-18 at 23 08 56](https://user-images.githubusercontent.com/1446331/118724249-a8818900-b82e-11eb-88a8-4218c22c094b.png)

![Screenshot 2021-05-18 at 23 08 59](https://user-images.githubusercontent.com/1446331/118724253-a9b2b600-b82e-11eb-883e-598efd730dbd.png)

In my code I'm doing a simple test to see if the PDF exists or not:

```html
{{ if pdf }}
      <dt>Assets</dt>
      <dd>
        <a href="{{ pdf }}" target="_blank">
          {{ slug }}.pdf
        </a>
      </dd>
{{ /if }}
```

## Environment
Statamic 3.1.13 Pro
Laravel 8.41.0
PHP 8.0.3
statamic/ssg 0.6.0

**Comment by tao** [comment]
I should add that yes I have run `php please stache:refresh`

**Comment by tao** [comment]
I have created an example repo: [statamic-translate-assets](https://github.com/tao/statamic-translate-assets). If you switch to the french site you'll see the same image as the english page when the entry value is null.

I found another issue when creating this example: not sure if it's intentional or not but when the asset field in the blueprint is not enabled as `localizable`...  you are still able to delete the image on the translation if you hover over the image.
